src/main/java/com/suse/saltstack/netapi/config/ClientConfig.java

```diff
@@ -12,6 +12,28 @@
     public static final Key<URI> URL = new Key<>(URI.create("http://localhost:8000"));
     public static final Key<String> TOKEN = new Key<>();
 
+    /**
+     * Timeout in milliseconds until a connection is established.
+     * A timeout of zero is interpreted as an infinite timeout.
+     * A negative value is interpreted as undefined (system default).
+     * Default value is 10000ms (10s)
+     * 
+     * @see HttpClientConnection.request(String)
+     * @see JDKConnection.request(String, String)
+     */
+    public static final Key<Integer> CONNECT_TIMEOUT = new Key<>(10000);
+
+    /**
+     * Timeout in milliseconds for waiting for data.
+     * A timeout of zero is interpreted as an infinite timeout.
+     * A negative value is interpreted as undefined (system default).
+     * Default value is 10000ms (10s)
+     * 
+     * @see HttpClientConnection.request(String)
+     * @see JDKConnection.request(String, String)
+     */
+    public static final Key<Integer> SOCKET_TIMEOUT = new Key<>(10000);
+
     // Proxy settings
     public static final Key<String> PROXY_HOSTNAME = new Key<>();
     public static final Key<Integer> PROXY_PORT = new Key<>(3128);
```

Comment:
This port (3128) was chosen as the default PROXY_PORT since it is the default port used by some HTTP proxy servers, e.g. squid. Making this a constant? Why not, but no need to discuss this here since it is very unrelated to this patch.